src/backend/cuda/platform.cpp

```diff
@@ -37,7 +39,18 @@
 
 using namespace std;
 
+
 namespace cuda {
+
+// check with reviewers : perhaps to be moved to a static of DeviceManager
+std::shared_ptr<spdlog::logger> logger = common::loggerFactory("cuda");
+
+#if defined(AF_WITH_LOGGING)
+spdlog::logger* getLogger() {
+  return logger.get();
+}
+#endif
```

Comment:
`getLogger` should be member function of deviceManager.

You can refer to `src/backend/common/MemoryManager.hpp` and `src/backend/common/MemoryManagerImpl.hpp` to see how it is done.

Reply:
done